### PR TITLE
cephadm: pin pyfakefs version

### PR DESCRIPTION
**Pull request by nizamial09**
I saw some cephadm tox failures, while checking I saw there is an update in the pyfakefs dep yesterday. So I guess that caused the issue?

https://jenkins.ceph.com/job/ceph-pull-requests/104991/consoleFull#-1067535055c212b007-e891-4176-9ee7-2f60eca393b7

```
plugins: pyfakefs-5.0.0
collected 187 items

tests/test_cephadm.py .................................................. [ 26%]
....................................FF.Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x7fe71e0527a0>
Traceback (most recent call last):
  File "/usr/lib/python3.10/tempfile.py", line 589, in __del__
  File "/usr/lib/python3.10/tempfile.py", line 582, in close
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm/.tox/py3/lib/python3.10/site-packages/pyfakefs/fake_filesystem.py", line 5121, in close
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm/.tox/py3/lib/python3.10/site-packages/pyfakefs/fake_filesystem.py", line 5157, in flush
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm/.tox/py3/lib/python3.10/site-packages/pyfakefs/helpers.py", line 339, in getvalue
ValueError: I/O operation on closed file.
Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x7fe71e0527a0>
```

Signed-off-by: Nizamudeen A <nia@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
```
176/282 Test  #22: run-tox-cephadm ...........................   Passed  105.78 sec
```

File: src/cephadm/tox.ini [hunk 1]
```diff
@@ -34,7 +34,7 @@ addopts =
 [testenv]
 skip_install=true
 deps =
-  pyfakefs
+  pyfakefs==4.7.0
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/cephadm/tox.ini:37]: i mostly prefer a strict pinning. But if its avoiding an update to a major version then I guess its okay.

**Comment by adk3798** [comment]
> see #48421 . i'd pin pyfakefs to v5.0 instead with a root cause analysis.

@tchaikov Messaged with @nizamial09 and we both like the approach in that PR. I think we can go with that as long as the tests pass. Any reason for it still being in draft? I figure we can pin to 5.0 after your change.

**Comment by tchaikov** [comment]
> > see #48421 . i'd pin pyfakefs to v5.0 instead with a root cause analysis.
> 
> @tchaikov Messaged with @nizamial09 and we both like the approach in that PR. I think we can go with that as long as the tests pass. Any reason for it still being in draft? I figure we can pin to 5.0 after your change.

@adk3798 hi Adam, because it didn't passed the "make check" run yet by then. now that it passed, i am marking it ready!

**Comment by nizamial09** [comment]
@tchaikov yours makes more sense so I'll close it out wrt to #48421. And can we get it merged faster? @adk3798 some PRs are currently blocked by it.

**Pull request closed without merging**